src/webview/notifications.ts

```diff
@@ -37,12 +37,22 @@ export const notificationsClassDefinition = `(() => {
     constructor(title = '', options = {}) {
       this.title = title;
       this.options = options;
-      window.ferdium.displayNotification(title, options)
-        .then(() => {
-          if (typeof (this.onClick) === 'function') {
-            this.onClick();
-          }
-        });
+      try{
+        window.ferdium.displayNotification(title, options)
+          .then(() => {
+            if (typeof (this.onClick) === 'function') {
+              this.onClick();
+            }
+          });
+      } catch(error) {
+	        this.options.onClick = null;
+          window.ferdium.displayNotification(title, options)
+            .then(() => {
+              if (typeof (this.onClick) === 'function') {
+                this.onClick();
+              }
+            });      
+      }
```

Comment:
I think you got confused a bit with the try catch and then/catch blocks, With your code, the same method would be called in both try and catch. 

What do you think about this:

```suggestion
try {
  window.ferdium.displayNotification(title, options);

  if (typeof this.onClick === 'function') {
    this.onClick();
  }
} catch {
  this.options.onClick = null;
}
```

Reply:
I understand why it is seen as a bit repetitive, however I think it may be necessary.  IIRC, 'this.onClick' will still return as being typeof function even in the failure scenario, but when the displayNotifications function is called the "options" object seems to fail to be copied for some reason (something seemingly to do with the onClick function).  My point is, the error occurs further down the line so we can't just assume that it showing as typeof function is enough for it to succeed.

Reply:
Just checked, and as I said, it seems that typeof returns the correct value even if it will then go on to fail.  See screenshot below.  First debug console command was run at the breakpoint, and then when execution is resumed the error is thrown.

![image](https://github.com/ferdium/ferdium-app/assets/622379/19d60d78-c8f6-415b-b397-c8451171ed58)

Reply:
Thanks for the thorough investigation! It seems it has to be written that way, so fine with continuing. 

Still would like to defer to @vraravam

Reply:
I am not enough of a js developer to comment on this nuance. I will defer to you and other core committers for the decision.

Reply:
I'm no JS expert either. That said, this seems to mostly fix a significant issue even if it's a bit of a dirty hack, so I hope you will approve it.